// callHealth/callHealth.js
var GetHealthAPI = function() {
    var https = require('https');
	/**
 	* GET Activity Data from Health API Current Time - 5 mins
 	*/


// Configure the application with you generated App ID and App Secret
var HUMANAPI_APP_ID     =  '3e098d19e42de61584e1bac7ff321a7385edd094';
var HUMANAPI_APP_SECRET =  'd5f5725383e96b2d6fd8db5897b8535b2ac69ca6' ;

var moment = require('moment');

//moment.duration.subtract(5, 'm' );
//moment.subtract(5, 'm' );
moment.duration().subtract(5, 'm');

var curTimeMinusFive = moment.utc().format('YYYYMMDDThhmmssZ');

console.info('Current Time:\t' + curTimeMinusFive) ;

//Current Time Stamp
var lastUpdateTs = { 'lastDataRetrievalTime' : curTimeMinusFive};

//Get Last retrived timestamp from the database

//var path = '/v1/apps/' + HUMANAPI_APP_ID + '/users/activities/summaries' + '&updated_since=' + curTimeMinusFive ; 
var path = '/v1/apps/' + HUMANAPI_APP_ID + '/users/activities/summaries' ;

var postheaders = {
    'Accept': 'application/json'
};

var optionsget = { 
    auth : '96df20174a8f47178519f08cd49153b8f221b1bc:',
    host : 'api.humanapi.co', // here only the domain name
    port : 443,
    //path : '/v1/apps/3e098d19e42de61584e1bac7ff321a7385edd094/users/activities/summaries', // the rest of the url with parameters if needed
    path : path,
    method : 'GET', // do GET
    headers : postheaders // do GET
};
 
console.info('Options prepared:');
console.info(optionsget);
console.info('Do the GET call');
 
// do the GET request
var reqGet = https.request(optionsget, function(res) {
    console.log("statusCode: ", res.statusCode);
    // uncomment it for header details
//  console.log("headers: ", res.headers);
 
 
    res.on('data', function(d) {
	//Store last retrieval timestamp in database Database and 
        console.info('GET result:\n');
        process.stdout.write(d);
        console.info('\n\nCall completed');
    });
 
});
 
reqGet.end();
reqGet.on('error', function(e) {
    console.error(e);
});
};

exports.GetHealthAPI = GetHealthAPI ;
